src/ncp/example_vendor_hook.cpp

```diff
@@ -35,6 +35,7 @@
 namespace ot {
 namespace Ncp {
 
+#if OPENTHREAD_ENABLE_NCP_VENDOR_HOOK
```

Comment:
Can we wrap all source with this feature flag?  In particular, move this between the doxygen comment and `#include` above?  Similarly, move the `#endif` below to the last line?